05-Mining-Protocol.md

```diff
@@ -1,106 +1,195 @@
 
-The size of search space for an extended channel is `2^(NONCE_BITS+VERSION_ROLLING_BITS+extranonce_size*8)` per `nTime` value.
+This separation vastly simplifies the protocol implementation for clients that don’t support Extended Jobs, as they only need to implement the subset of protocol messages related to Standard Jobs (see Mining Protocol Messages for details).
 
-### 5.1.3 Group Channels
+Additionally, a Job (either Standard or Extended) also could be potentially labeled as a **Future Job** and/or **Custom Job**.
 
-Standard channels opened within one particular connection can be grouped together to be addressable by a common communication group channel.
+### 5.1.1 Standard Jobs
+
+Standard Jobs are restricted to fixed Merkle Roots, where the only modifiable bits are under the `version`, `nonce`, and `nTime` fields of the block header.
+
+We call this header-only mining (HOM), and it is the smallest assignable unit of search space by the protocol.
+
+The size of the search space for one Standard Job, given a fixed `nTime` field, is `2^(NONCE_BITS + BIP320_VERSION_ROLLING_BITS) = ~280Th`, where `NONCE_BITS = 32` and `BIP320_VERSION_ROLLING_BITS = 16`.
+This is a guaranteed space before `nTime` rolling (or changing the Merkle Root by sending a new Job).
+
+Standard Jobs are distributed via the [`NewMiningJob`](#5315-newminingjob-server---client) message, which can ONLY be sent via [Standard Channels](#521-standard-channels).
+
+Note that Mining Devices with hashrate above 280 TH/s will outrun the true time when rolling the `nTime` field. This is not necessarily a problem, as long as the Job remains valid within the limits defined by the `MAX_FUTURE_BLOCK_TIME` constant of the Bitcoin protocol (2 hours).
+
+Depending on the hashrate of the Mining Device, it must receive new Standard Jobs (with a different Merkle Root) frequently enough in order to avoid rolling `nTime` too much ahead of the true time (which could result in rejected work).
+
+![](./img/standard_job.png)
+
+### 5.1.2 Extended Jobs
+
+Extended Jobs allow rolling Merkle Roots, giving extensive control over the search space so that they can implement various advanced use cases such as:
+- translation between Sv1 and Sv2 protocols
+- difficulty aggregation
+- search space splitting
+
+Extended Jobs are distributed via the [`NewExtendedMiningJob`](#5416-newextendedminingjob-server---client) message, which can be sent via:
+- [Group Channels](#523-group-channels)
+- [Extended Channels](#522-extended-channels)
+
+![](./img/extended_job.png)
+
+#### 5.1.2.1 Extended Extranonce
+
+Downstream and Upstream are relative terms. There could theoretically be multiple nested layers of Stratum Nodes (e.g.: Proxies) between a Work-Providing Node and a Mining Device.
+
+So the protocol establishes the notion of **Extended Extranonce**, which is a fundamental component of Extended Jobs.
+
+The Extended Extranonce is an array of 32 bytes, split into three different areas:
```

Comment:
note: we need to verify this

I'm not 100% confidence this is limited to 32 bytes

Reply:
cc @GitGab19

Reply:
there is a limitation of 100 bytes to `scriptSig`, but extranonce is not not necessarily limited to 32 bytes